File: 2022/python/day1/day1p2.py
#!/usr/bin/python

from lib import *

import functools

def main():
    lines = get_elf_calories("input")

    if len(lines) >= 3:
        rc = functools.reduce(lambda x, y: x+y, lines[0:3])
        print(rc)
    else:
        print("Error: No calories")


if __name__ == "__main__":
    main()

File: 2022/cpp/day1/src/day1p2.cpp
#include "lib.h"

#include <iostream>
#include <numeric>

int main(void) {
    auto elf_cals = get_elf_calories("input");

    if (elf_cals.size() >= 3) {
        auto rc = std::accumulate(elf_cals.begin(), elf_cals.begin() +3, 0);
        std::cout << rc << "\n";
    } else {
        std::cout << "Error: no calories" << "\n";
    }

    return 0;
    
}
